ddtrace/mocktracer/mocktracer.go

```diff
@@ -79,12 +95,23 @@ func (t *mocktracer) FinishedSpans() []Span {
 func (t *mocktracer) Reset() {
 	t.Lock()
 	defer t.Unlock()
+	t.openSpans = nil
 	t.finishedSpans = nil
 }
 
+func (t *mocktracer) addOpenSpan(s Span) {
+	t.Lock()
+	defer t.Unlock()
+	if t.openSpans == nil {
```

Comment:
Sorry, my initial comment might've not been clear. Can we please get rid of this method? Instead, we can:
* Make the map in `Start`
* Simply lock, add to the map, and unlock in `StartSpan`

It's a bit easier to read without the need of additional methods and concepts.

Thanks!

Reply:
Done